### ISSUE TITLE
Issue with LabelBinarizer

### ISSUE THREAD

**Issue by Dovermore** [ageron/handson-ml#146]
The first part is the newer version of sklearn (ver1.9) no longer support using `LabelBinarizer` in `Pipeline`.
I ran into issue of `fit_transform() takes 2 positional arguments but 3 were given with LabelBinarizer`
One way around this is to create customized labelbinarizer using get_dummies in pandas.
But when doing so, I ran into another problem that the
 `some_data_prepared = full_pipeline.transform(some_data)`
don't give the desired dimension (16), rather give dim 14.  Reason being the first 5 sets of data only contain three types of location.
Which is solved by:
https://stackoverflow.com/questions/37425961/dummy-variables-when-not-all-categories-are-present

**Comment by Mani256** [comment]
Hi Dovemore, 

I ran into the same problems(with LabelBinarizer and with some_data_prepared). I managed the LabelBinarizer with custom LabelBinarizer version, but it is failing while running the full pipeline on some_data. the cause is what you mentioned above...only three types of locations in the first 5 sets of data.

Is it possible for you to share the code? it would be very helpful. Thank you.

**Comment by ganeshth** [comment]
Hi @ageron would you be able to help. I'm unable to move further in the book because of this. Thank you

**Comment by Dovermore** [comment]
@ganeshth [You can check my implementation here.](https://github.com/Dovermore/handson-ml/blob/master/chapter_2_exer/chapter_2.ipynb) My code might be bit hard to read. try search `CustomLabelBinarizer` to find the segment of code you want. Also, I've modified further and used it in multiple other pipelines in [here](https://github.com/Dovermore/handson-ml/blob/master/chapter_3_exer/question_3.ipynb) and few other places. It now works fine for me.

**Comment by ganeshth** [comment]
@Dovermore Many thanks, I will modify my code accordingly. 

You did the titanic dataset too. Great!!! that's next in my list of Machine Learning :grin:

**Comment by ganeshth** [comment]
@Dovermore This is actually #55 and is fixed and closed long back. The current notebook has the implementation and warning messages. I took that code. I suggest you to go through that issue comments as there is a lot of discussion about what to use and what not to. 

Finally!!! you can close the issue as this has been implemented.

**Comment by ageron** [comment]
Hi @Dovermore and @ganeshth ,
With the winter holidays and tons of work, I didn't answer as quickly as I would have liked, my apologies for the delay.  I'm glad you found the solution in the meantime!
Cheers,
Aurélien
